README.md

```diff
@@ -1,3 +1,312 @@
+
+```yaml
+name: Minimal GPU EC2 runner test
+on:
+  workflow_dispatch:
+permissions:
+  id-token: write  # Required for AWS OIDC authentication
+  contents: read   # Required for actions/checkout
+jobs:
+  ec2:
+    uses: Open-Athena/ec2/.github/workflows/runner.yml@main
+    secrets: inherit
+  gpu-test:
+    needs: ec2
+    runs-on: ${{ needs.ec2.outputs.instance }}
+    steps:
+    - run: nvidia-smi  # g4dn.xlarge!
+```
+
+That's it! The EC2 instance starts, runs your job, and automatically terminates when done.
+
+## Full Example
+
+For more control over instance configuration:
+
+```yaml
+name: GPU Tests
+
+on: [push, pull_request]
+
+jobs:
+  ec2:
+    uses: Open-Athena/ec2/.github/workflows/runner.yml@main
+    secrets: inherit  # Requires `AWS_ROLE`, `GH_SA_TOKEN`
+    with:
+      aws_instance_type: "g4dn.xlarge"  # Optional, defaults to g4dn.xlarge
+
+  test:
+    needs: ec2
+    runs-on: ${{ needs.ec2.outputs.instance }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Verify GPU availability
+        run: nvidia-smi
+
+      # No stop job needed - instance self-terminates!
+```
+
+## Configuration
+
+### Workflow Inputs
+
+| Input | Description | Default |
+|-------|-------------|---------|
+| `aws_instance_type` | EC2 instance type | [`g4dn.xlarge`](https://instances.vantage.sh/aws/ec2/g4dn.xlarge) (cheapest GPU instance) |
+| `aws_image_id` | AMI ID | `ami-00096836009b16a22` (Deep Learning AMI) |
+| `aws_home_dir` | Home directory path | `/home/ubuntu` |
+| `aws_key_name` | EC2 key pair name for SSH access | - |
+| `aws_security_group_id` | Security group ID (must allow SSH if using) | - |
+| `ssh_pubkey` | Additional SSH public key to authorize | - |
+| `shutdown_poll_wait` | Minutes to wait for runner setup before monitoring | `3` |
+| `poll_interval` | Seconds between runner process checks | `15` |
+
+### Environment Variables
+
+Set these as organization or repository variables for defaults:
+
+- `EC2_IMAGE_ID` - Default AMI ID
+- `EC2_INSTANCE_TYPE` - Default instance type
+- `EC2_KEY_NAME` - Default SSH key pair name
+- `EC2_SECURITY_GROUP_ID` - Default security group ID
+
+Priority: workflow inputs > environment variables > hardcoded defaults
+
+## How It Works
+
+1. The workflow starts an EC2 instance using the specified configuration
+2. A GitHub Actions runner is automatically installed and registered
+3. Your job runs on the EC2 instance
+4. A systemd service monitors the runner process
+5. When the runner stops (job completes), the instance self-terminates automatically
+
+## Security
+
+### Fork Protection
+This workflow includes built-in protection against unauthorized EC2 launches from forked repositories. EC2 instances will only start for:
+- Direct pushes to your repository
+- Manual workflow dispatches
+- Pull requests from branches within your repository (not forks)
+- Pull requests from forks that have the approval label (default: `gpu`) applied by a maintainer
```

Comment:
Gotta stop those crypto miners.

Reply:
This comment was out of date, thanks for bringing that to my attention! Fixed.